WeakAuras/Prototypes.lua

```diff
@@ -1367,57 +1381,58 @@ local function AddUnitEventForEvents(result, unit, event)
   end
 end
 
+local unitHelperFunctions = {
+  UnitChangedForceEvents = function(trigger)
+    local unit = trigger.unit
+    local events = {}
+    if WeakAuras.multiUnitUnits[trigger.unit] then
+      for unit in pairs(WeakAuras.multiUnitUnits[trigger.unit]) do
+        tinsert(events, {"UNIT_CHANGED_" .. unit, unit})
+      end
+    else
+      if unit then
+        tinsert(events, {"UNIT_CHANGED_" .. unit, unit})
+      end
+    end
```

Comment:
variable `unit` is overwritten in the for loop using same name
`unit` nil check in the else block should be done before the test of `WeakAuras.multiUnitUnits[trigger.unit]`

Reply:
The else block is never entered after the loop. But the code looks a bit convulted, I'll simplify it.